### ISSUE TITLE
Game Tally Mismatch

### ISSUE THREAD

**Issue by tarantula2107** [ebshimizu/stats-of-the-storm#206]
My match history only comes up with a single Yrel game while the Player Details page for the only account I own Yrel on shows two games on her. However, I have two incomplete Dragon Shire games on 28 June so one of them could be her. I think you should try to find a way to discount incomplete games from the stats because the replays won't upload at all.

**Comment by ebshimizu** [comment]
If a match is marked as incomplete in the parser, it is not imported into the database (incomplete means that it can't determine the winner for whatever reason, usually due to a truncated replay), so those shouldn't be included

It's possible that your date filters are set up in a way that excludes some matches. The end date in the match filters is _exclusive_, meaning that if you search for games starting on June 21 to June 29, it will include games from June 21 at 12am to June 29 at 12am, so any games played on the day off the 28th will not show up. I am considering changing this to an inclusive bound.

Try changing the date filters, and let me know if that resolves the issue. If not, I'll need a little more detail on how to reproduce the error.

**Comment by tarantula2107** [comment]
Current date filter is 1/1/16-7/9/18. The bug seems to be it thinking I played Yrel on my high-level (old account as illustrated below when I don't even own her there yet. To reproduce to error, I, rather simply, just toggled the player ID in the appropriate search box several times before taking the screens below.
![yrel new acc sots](https://user-images.githubusercontent.com/40201918/42449663-8d9442ac-834f-11e8-9e43-7a379512cc04.png)
![yrel old acc sots](https://user-images.githubusercontent.com/40201918/42449664-8da5bd48-834f-11e8-808e-c7170815e379.png)

**Comment by ebshimizu** [comment]
I'm not sure there's actually not an error here.

The player/hero filters work with an OR operation by default. So if you filter by your ID and select a Hero (Yrell), it will select all games that you were in or that had a Yrell in. You can hover over the hero portrait to see who played them.

If you want to restrict the filter such that it only shows you matches where you played the selected hero, toggle the check box under the hero filter.

**Comment by tarantula2107** [comment]
I'm relatively sure there's an error here because I haven't even played Yrel on Tarantula#11354 (lower image) because (a.) I don't own her on that account and (b.) I'm reasonably sure she hasn't been free-to-play yet (missed it there if so). To further prove the error, I've attached Tarantula#11361's (upper image) HotSLogs info below (third game down). I no longer upload Tarantula#11354's replays to HotSLogs to make sure my available MMR is as close to my internal MMR, and I'm missing a few full replays on that account.

![yrel hotslogs](https://user-images.githubusercontent.com/40201918/42467624-62eebb5a-8380-11e8-9df5-31a73f566861.png)

**Comment by ebshimizu** [comment]
Yeah so again in the lower image you are searching for matches where one of two things are occurring: 
* the match includes Tarantula#11354
* OR the match includes someone that played Yrell

This means that if your old account played Yrell in that match (True) it will appear, because the filters are Tarantula#11354 OR [Played Yrell]. Try removing the Yrell filter and seeing if the match still show up. If so, send me the database and I'll take a look.

**Comment by tarantula2107** [comment]
Point. You know the program better than me. However, the image below should just about prove there's an error somewhere. I should have attached this much earlier.

![yrel new acc sots01](https://user-images.githubusercontent.com/40201918/42470898-e43d6626-8389-11e8-8921-0a0222fc0e6a.png)

**Comment by ebshimizu** [comment]
Hmmm and that’s from the new profile?
Go ahead and zip up your database folder and send to me. That should only be happening if the same internal id (different than battletag) was used in the game. 

I’ll take a look as soon as I can, but will be prioritizing support for the third ban patch on Tuesday.

**Comment by ebshimizu** [comment]
database received, will look as soon as I can. Expect later this week or over the weekend (new patch takes priority over everything)

**Comment by tarantula2107** [comment]
Okay. Although, the old one is 11354, with the new one being 11361.

**Comment by ebshimizu** [comment]
There's currently an open issue for merging battletags #169 
This has become increasingly requested, so I'll be taking a look at how feasible it is in the coming weeks.
